### PR TITLE
NUI-4910 Tests Review and Improvements for menu component

### PR DESCRIPTION
**Pull request by elenates**
- deleted several e2e tests which were duplicated by visual tests,
- added a component to visual test to reduce time of test running,
- renamed one section to use it in visual test instead of e2e test.

### CODE REVIEW

File: packages/bits/demo/src/components/demo/menu/menu-visual-test/menu-visual-test.component.ts [hunk 1]
```diff
@@ -40,6 +40,13 @@ export class MenuVisualTestComponent {
             {title: "Link menu item", itemType: "link", url: "#button", disabled: true},
             {title: "Export PDF", itemType: "link", icon: "export-pdf", url: "#button"},
         ]},
+        {
+            header: "section 2 title", itemsSource: [
```
Comment (Blakesalot) [inline packages/bits/demo/src/components/demo/menu/menu-visual-test/menu-visual-test.component.ts:44]: Newline preferred here. Also, do you mind updating the format of the other objects as well at lines 11, 19, and 32?

```suggestion
            header: "section 2 title",
            itemsSource: [
```

**Review by Blakesalot — DISMISSED** [review]

**Review by Blakesalot — DISMISSED** [review]

**Review by madelyneh — DISMISSED** [review]

**Review by Blakesalot — APPROVED** [review]

**Review by madelyneh — APPROVED** [review]

File: packages/bits/spec/components/menu/menu.e2e.ts [hunk 2]
```diff
@@ -30,82 +23,7 @@ describe("USERCONTROL Menu", () => {
...
-            const disabledOptionItem = menu.getMenuItemByContainingText("Disabled option");
-            const checkbox: CheckboxAtom = Atom.findIn(CheckboxAtom, disabledOptionItem.getElement());
-            expect(await checkbox.isChecked()).toBe(false);
-        });
-        it("should locate disabled switch", async () => {
-            const disabledSwitchItem = menu.getMenuItemByContainingText("Disabled switch");
-            const switchAtom = Atom.findIn(SwitchAtom, disabledSwitchItem.getElement());
-            expect(await switchAtom.isOn()).toBe(false);
-        });
-    });
```
Comment (ivankozlovskyi) [inline packages/bits/spec/components/menu/menu.e2e.ts:91]: It seems like there are no such cases in visual tests

File: packages/bits/spec/components/menu/menu.e2e.ts [hunk 3]
```diff
@@ -130,22 +47,7 @@ describe("USERCONTROL Menu", () => {
...
 
     describe("> key navigation", () => {
-        const checkElementsActiveState = async () => {
-            await menu.toggleMenu();
-            await browser.actions().sendKeys(protractor.Key.ARROW_DOWN).perform();
-            expect(await menu.getMenuItemByIndex(0).isActiveItem()).toBe(true);
-            await browser.actions().sendKeys(protractor.Key.ARROW_DOWN).perform();
-            expect(await menu.getMenuItemByIndex(1).isActiveItem()).toBe(true);
-            await browser.actions().sendKeys(protractor.Key.ARROW_DOWN).perform();
-            expect(await menu.getMenuItemByIndex(2).isActiveItem()).toBe(true);
```
Comment (ivankozlovskyi) [inline packages/bits/spec/components/menu/menu.e2e.ts:140]: I think that only last one should be removed

File: packages/bits/spec/components/menu/menu.e2e.ts [hunk 4]
```diff
@@ -202,14 +107,12 @@ describe("USERCONTROL Menu", () => {
                 });
 
                 it("should select and close menu when selecting menu action item", async () => {
-                    await menu.toggleMenu();
```
Comment (ivankozlovskyi) [inline packages/bits/spec/components/menu/menu.e2e.ts:205]: have you checked that menu is opened in this case, it is possible that those tests where  "await menu.toggleMenu();" was deleted become false positive

Comment (elenates) [reply packages/bits/spec/components/menu/menu.e2e.ts:205]: Discussed offline - "await menu.toggleMenu();" was moved to _beforeEach_

File: packages/bits/spec/components/menu/menu.e2e.ts (hunk 2 above)
Comment (elenates) [reply packages/bits/spec/components/menu/menu.e2e.ts:91]: Discussed offline - _Disabled switch_, _Disabled action_, _Disabled link_ and _Disabled option_ are verified in another visual tests.

File: packages/bits/spec/components/menu/menu.e2e.ts (hunk 3 above)
Comment (Blakesalot) [reply packages/bits/spec/components/menu/menu.e2e.ts:140]: I think Ivan may have meant that the test itself shouldn't be deleted, but we should keep it and just delete lines 139 and 140. Do I have that right, Ivan? But, maybe there is a visual test that covers the whole scenario?

**Comment by Blakesalot** [comment]
@elenates, I just noticed that the browser gets refreshed with every test. I wonder if it would be an easy task to put the preparBrowser in a beforeAll instead of a beforeEach. Each test might have to clean up after itself, e.g. restore a menu to its default state somehow, or maybe there could be some generic cleanup in an afterEach. If it's too much trouble, no worries. I was just thinking of a way to speed the test suite up by avoiding unnecessary page refreshes.

File: packages/bits/spec/components/menu/menu.e2e.ts (hunk 3 above)
Comment (elenates) [reply packages/bits/spec/components/menu/menu.e2e.ts:140]: Hi Blake, sorry, we discussed it offline with Ivan, and I didn't write it. Yes, the visual test covers the whole scenario anyway.

**Comment by elenates** [comment]
@Blakesalot, thank you for your comment, that a great idea. I am working on that, and looks like we can speed the test up from 8 to 3 secs.

**Pull request merged**